specs/ExtendedProcessFailed.md

```diff
@@ -0,0 +1,671 @@
+                // A frame-only renderer has exited unexpectedly. Check if reload is needed.
+                wil::com_ptr<ICoreWebView2FrameInfoCollection> impactedFrames;
+                wil::com_ptr<ICoreWebView2FrameInfoCollectionIterator> iterator;
+                CHECK_FAILURE(args2->get_ImpactedFramesInfo(&impactedFrames));
+                CHECK_FAILURE(impactedFrames->GetIterator(&iterator));
+
+                BOOL hasCurrent = FALSE;
+                while (SUCCEEDED(iterator->HasCurrentFrameInfo(&hasCurrent)) && hasCurrent)
+                {
+                    wil::com_ptr<ICoreWebView2FrameInfo> frameInfo;
+                    CHECK_FAILURE(iterator->GetCurrentFrameInfo(&frameInfo));
+
+                    wil::unique_cotaskmem_string nameRaw;
+                    wil::unique_cotaskmem_string sourceRaw;
+                    CHECK_FAILURE(frameInfo->get_Name(&nameRaw));
+                    CHECK_FAILURE(frameInfo->get_Source(&sourceRaw));
+                    std::wstring source = sourceRaw.get();
+
+                    // Content from our app uses a mapped host name.
+                    const std::wstring mappedAppHostName = L"https://appassets.example/";
+                    if (source.compare(0, mappedAppHostName.length(), mappedAppHostName) == 0)
+                    {
+                        int button = MessageBox(
+                            m_appWindow->GetMainWindow(),
+                            L"Browser render process for app frame exited unexpectedly. "
+                            L"Reload page?",
+                            L"App content frame unresponsive", MB_YESNO);
+                        if (button == IDYES)
+                        {
+                            CHECK_FAILURE(m_webView->Reload());
+                        }
+                        break;
+                    }
+
+                    BOOL hasNext = FALSE;
+                    CHECK_FAILURE(iterator->MoveNext(&hasNext));
+                }
+            }
+            else
+            {
+                // Show the process failure details. Apps can collect info for their logging
+                // purposes.
+                COREWEBVIEW2_PROCESS_FAILED_REASON reason;
+                wil::unique_cotaskmem_string processDescription;
+                int exitCode;
+
+                CHECK_FAILURE(args2->get_Reason(&reason));
+                CHECK_FAILURE(args2->get_ProcessDescription(&processDescription));
+                CHECK_FAILURE(args2->get_ExitCode(&exitCode));
+
+                std::wstringstream message;
+                message << L"Process kind:\t"        << ProcessFailedKindToString(failureKind) << L"\n"
+                        << L"Reason:\t"              << ProcessFailedReasonToString(reason)    << L"\n"
+                        << L"Exit code:\t"           << std::to_wstring(exitCode)              << L"\n"
+                        << L"Process description:\t" << processDescription.get()               << std::endl;
```

Comment:
This line crashes if there is no description, because the platform produces a null pointer if there is no description, and inserting a null string pointer into a stream is undefined behavior.

Maybe the platform should return a non-null pointer to an empty string?

Reply:
Updated to empty string.